doc/API.md

```diff
@@ -991,6 +992,29 @@ lists all files in the download queue.
 }
 ```
 
+#### /renter/downloads/clear [POST]
+
+Clears the download history of the renter for a range
+of unix time stamps.  Both parameters are optional, if
+no parameters are provided, the entire download history
+will be cleared.  To clear a single download, provide the
+timestamp for the download as both parameters.  Since
+download history is displayed from newest to oldest, providing
+only the start parameter will clear all downloads before, or
+older than the start timestamp.  Conversely, providing only
+the end parameter will clear all downloads after, or
+newer than the end timestamp.
```

Comment:
This sounds backwards. If I specify `clear?start=10`, then it should clear everything *newer* than (after) 10. If I specify `clear?end=10`, it should clear everything *older* than (before) 10.

also, we should specify that the time stamp is in nanoseconds, and that the nanosecond value is reported in `/downloads`. I'm not sure how well other languages support nanosecond timestamps, so in some cases the easiest thing to do might be to simply copy the timestamps from the starting and ending downloads in the range you want to clear.

Reply:
I can rephrase it a little bit, but the intention is that the start and end define the range and since we display to the user the downloads from newest to oldest it makes sense to think of it as where do I start clearing the download history and where to I end clearing the download history.  So clear from the start onward or up to the end. Maybe just removing before and after would make it clearer, so something like this:

`Since the download history is displayed from newest to oldest, providing
only the start parameter will clear all downloads from the start timestamp
onward, which would be all downloads older than the start timestamp.  
Conversely, providing only the end parameter will clear all downloads up to
the end timestamp, which would be all downloads newer than the end timestamp.
`

Reply:
I think I understand, but IMO it's surprising that the "start" timestamp would come after the "end" timestamp. It's true that when presented to the user, downloads are usually ordered newest to oldest, but that's a presentation choice. The user isn't the one crafting the API calls, the app developer is, so we should prioritize the developer and try to meet their expectations.

One possible solution would be to rename `start`/`end`/ to `oldest`/`newest`. Then the meaning is unambiguous.